providers/implementations/kdfs/hkdf.c

```diff
@@ -180,6 +186,34 @@ static size_t kdf_hkdf_size(KDF_HKDF *ctx)
     return sz;
 }
 
+#ifdef FIPS_MODULE
+static int fips_hkdf_digest_check_passed(KDF_HKDF *ctx)
+{
+    OSSL_LIB_CTX *libctx = PROV_LIBCTX_OF(ctx->provctx);
+    const EVP_MD *md = ossl_prov_digest_md(&ctx->digest);
+    /*
+     * Perform digest check
+     *
+     * HKDF is a TwoStep KDF defined in SP 800-56Cr2. According to section 7,
+     * the valid hash functions are specified in FIPS 180 and FIPS 202.
+     * However, it only lists SHA-1, SHA-2 and SHA-3 in the table in section
+     * 5.2. ACVP also only lists the same set of hash functions.
+     */
+    int digest_unapproved = (md != NULL)
```

Comment:
NULL check is done in derive so it is not needed here..

Reply:
Unneeded check removed.

Note that the check in the `fips_sskdf_digest_check_passed` function is untouched to avoid duplicate code in the `sskdf_derive` function.